globus_cli/services/transfer/bookmark/locate.py

```diff
@@ -0,0 +1,49 @@
+from uuid import UUID
+
+import click
+
+from globus_sdk.exc import TransferAPIError
+from globus_cli.safeio import safeprint
+from globus_cli.parsing import common_options
+from globus_cli.services.transfer.helpers import get_client
+
+
+@click.command('locate', help='Output <UUID>:<path> of a bookmark; useful in '
+                              'a subshell as input to another command')
+@common_options
+@click.argument('bookmark_id_or_name', required=True)
+def bookmark_locate(bookmark_id_or_name):
+    """
+    Executor for `globus transfer bookmark locate`
+    """
+
+    client = get_client()
+
+    # leading/trailing whitespace doesn't make sense for UUIDs and the Transfer
+    # service outright forbids it for bookmark names, so we can strip it off
+    bookmark_id_or_name = bookmark_id_or_name.strip()
+
+    try:
+        UUID(bookmark_id_or_name)  # raises ValueError if argument not a UUID
+        res = client.get_bookmark(bookmark_id_or_name.lower())
+
+    except (ValueError, TransferAPIError) as ex:
+        if isinstance(ex, TransferAPIError) and ex.code != 'BookmarkNotFound':
```

Comment:
Yes, we can totally do something different.

I pushed b9260ae for a revised structure. The revised structure has the bonus of being a little bit safer in the event that `client.get_bookmark(bookmark_id_or_name.lower())` raised a `ValueError`, in which case the previous structure would have silenced that error.

If you're happy with the revised structure, I can squash it.